### PR TITLE
Add some Slack notifications during the release process

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

NA

**Description**

- announce the beginning and the end of a release on Slack automatically
- send Slack notification when the first release step is ok
- add missing slack configuration on notify-on-failure command

NB: All slack notifications are currently sent on `api-management_team_notifications` channel, we will move some of them to the public channel as soon as we will be confident enough ;)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/add-release-announcement/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -719,6 +719,24 @@ jobs:
...
+                channel: C02JENTV2AX
+                event: always
+                custom: |
+                  {
+                    "blocks": [
+                      {
+                        "type": "section",
+                        "text": {
+                          "type": "mrkdwn",
+                          "text": "🚀 Starting << pipeline.parameters.graviteeio_version >> release"
```
Comment (phiz71) [inline .circleci/config.yml]: Maybe adding the dry-run information ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -932,6 +950,24 @@ jobs:
...
+                channel: C02JENTV2AX
+                event: always
+                custom: |
+                  {
+                    "blocks": [
+                      {
+                        "type": "section",
+                        "text": {
+                          "type": "mrkdwn",
+                          "text": "🎆 APIM - << pipeline.parameters.graviteeio_version >> released!"
```
Comment (phiz71) [inline .circleci/config.yml]: Maybe adding the dry-run information ?

**Review by ytvnr — APPROVED** [review]
good job

File: .circleci/config.yml [hunk 3]
```diff
@@ -830,6 +830,22 @@ jobs:
...
+                  channel: C02JENTV2AX
+                  event: pass
+                  custom: |
+                      {
+                        "blocks": [
+                          {
+                            "type": "section",
+                            "text": {
+                              "type": "mrkdwn",
+                              "text": "✅ APIM repository has been tagged.\nYou can continue the release by running: `npm run package_zip -- --version=<< pipeline.parameters.graviteeio_version >>`"
```
Comment (phiz71) [inline .circleci/config.yml]: Adapt the comment if in dry-run mode (or do not notify)

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -771,6 +775,28 @@ jobs:
...
+                        "blocks": [
+                          {
+                            "type": "section",
+                            "text": {
+                              "type": "mrkdwn",
+                              "text": "🚀 Starting << pipeline.parameters.graviteeio_version >> release"
+                            }
+                          }
+                        ]
+                      }
```
Comment (phiz71) [inline .circleci/config.yml]: wrong indentation

File: .circleci/config.yml [hunk 5]
```diff
@@ -984,6 +1030,28 @@ jobs:
...
+                        "blocks": [
+                          {
+                            "type": "section",
+                            "text": {
+                              "type": "mrkdwn",
+                              "text": "🎆 APIM - << pipeline.parameters.graviteeio_version >> released!"
+                            }
+                          }
+                        ]
+                      }
```
Comment (phiz71) [inline .circleci/config.yml]: wrong indentation

**Review by phiz71 — APPROVED** [review]

**Pull request merged**